controlplane/kubeadm/controllers/scale_test.go

```diff
@@ -287,6 +296,20 @@ func TestSelectMachineForScaleDown(t *testing.T) {
 			expectErr:        false,
 			expectedMachine:  clusterv1.Machine{ObjectMeta: metav1.ObjectMeta{Name: "machine-3"}},
 		},
+		{
+			name:             "when there is a single machine marked with delete annotation key in machine collection, it returns only that marked machine",
+			cp:               annotatedControlPlane,
+			outDatedMachines: internal.NewFilterableMachineCollection(m6, m7),
+			expectErr:        false,
+			expectedMachine:  clusterv1.Machine{ObjectMeta: metav1.ObjectMeta{Name: "machine-7"}},
+		},
+		{
+			name:             "when there are multiple machines marked with delete annotation key in machine collection, it returns the oldest marked machine first",
+			cp:               annotatedControlPlane,
+			outDatedMachines: internal.NewFilterableMachineCollection(m7, m8),
+			expectErr:        false,
+			expectedMachine:  clusterv1.Machine{ObjectMeta: metav1.ObjectMeta{Name: "machine-8"}},
+		},
```

Comment:
It would be good to also add test cases covering the expected behavior when an annotated machine is not in outdatedMachines but is part of annotatedControlPlane